File: resources/capability-bundle/.claude/skills/transfer-question-design/scripts/build-transfer-matrix.mjs
const payload = JSON.parse(process.argv[2] ?? '{}');
const moduleId = String(payload.module_id ?? '').trim();
if (!moduleId) throw new Error('module_id is required');

const dimensions = payload.dimensions ?? {
  audience: ['executive', 'peer'],
  objective: ['explain', 'defend'],
  seconds: [30, 60],
};

const matrix = [];
for (const audience of dimensions.audience ?? []) {
  for (const objective of dimensions.objective ?? []) {
    for (const seconds of dimensions.seconds ?? []) {
      matrix.push({ module_id: moduleId, audience, objective, seconds });
    }
  }
}

process.stdout.write(JSON.stringify({ matrix }));

File: resources/capability-bundle/.claude/skills/transfer-question-design/scripts/build-transfer-matrix.py
import json
import sys


payload = json.loads(sys.argv[1] if len(sys.argv) > 1 else "{}")
module_id = str(payload.get("module_id", "")).strip()
if not module_id:
    raise ValueError("module_id is required")

dimensions = payload.get("dimensions") or {
    "audience": ["executive", "peer"],
    "objective": ["explain", "defend"],
    "seconds": [30, 60],
}
matrix = [
    {
        "module_id": module_id,
        "audience": audience,
        "objective": objective,
        "seconds": seconds,
    }
    for audience in dimensions.get("audience", [])
    for objective in dimensions.get("objective", [])
    for seconds in dimensions.get("seconds", [])
]

sys.stdout.write(json.dumps({"matrix": matrix}, ensure_ascii=False, separators=(",", ":")))
